lib/ansible/module_utils/basic.py

```diff
@@ -2621,14 +2622,9 @@ def run_command(self, args, check_rc=False, close_fds=True, executable=None, dat
             msg = "Argument 'args' to run_command must be list or string"
             self.fail_json(rc=257, cmd=args, msg=msg)
 
-        shell = False
```

Comment:
Do not move this.  You may want to remove setting shell = True in the above conditional block, though.